packages/core/src/types.ts

```diff
@@ -222,7 +242,7 @@ export interface File {
 /**
  * Represents the successful result of parsing a file.
  */
-export interface OkParseResult {
+export interface OkParseResult<T extends object = object> {
```

Comment:
Making this generic allows devs to specify the type of the `ast` property to get intellisense completions.